pkg/kubelet/dockertools/docker.go

```diff
@@ -166,24 +166,19 @@ func matchImageTagOrSHA(inspected dockertypes.ImageInspect, image string) bool {
 		// No Tag or SHA specified, so just return what we have
 		return true
 	}
+
 	if isTagged {
-		hostname, _ := dockerref.SplitHostname(named)
-		// Check the RepoTags for an exact match
+		// Check the RepoTags for a match.
 		for _, tag := range inspected.RepoTags {
-			// Deal with image with hostname specified
-			if len(hostname) > 0 {
-				if strings.HasSuffix(image, tag) {
-					return true
-				}
-
-			} else {
-				if tag == image {
-					// We found a specific tag that we were looking for
-					return true
-				}
+			// An image name (without the tag/digest) can be [hostname '/'] component ['/' component]*
+			// Because either the RepoTag or the name *may* contain the
+			// hostname or not, we only check for the suffix match.
+			if strings.HasSuffix(image, tag) || strings.HasSuffix(tag, image) {
```

Comment:
Sorry for the drive by, I'm looking for things to add retest-not-required to...

but won't this give false positives? like "docker.io/foobar/myimage:1" and "bar/myimage:1" probably shouldn't match but I think they will with this code?